test/Interop/SwiftToCxx/structs/nested-structs-in-cxx.swift

```diff
@@ -0,0 +1,54 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend %s -enable-library-evolution -typecheck -module-name Structs -clang-header-expose-decls=all-public -emit-clang-header-path %t/structs.h
+// RN: %FileCheck %s < %t/structs.h
+
+// RUN: %check-interop-cxx-header-in-clang(%t/structs.h -DSWIFT_CXX_INTEROP_HIDE_STL_OVERLAY -std=c++17)
+
+public enum AudioFileType {
+	public enum WaveType {
+		case THIS
+		case THAT
+	}
+
+	public struct SubType {
+		public var id: Int
+	}
+
+	case CAF(SubType), WAVE(WaveType)
+}
+
+public struct RecordConfig {
+	public enum AudioFormat {
+		case PCM, ALAC, AAC
+	}
+
+	public struct Directory {
+		public var path: String?
+	}
+
+	public struct File {
+		public var type: AudioFileType = .CAF(AudioFileType.SubType(id: 42))
```

Comment:
Looks like tabs and spaces got mixed up here, I will fix this in the next iteration.